### PR TITLE
Setup a github actions workflow

### PR DESCRIPTION
**Pull request by kmadisa**
Adding a workflow file to set up github actions to run the automated unit tests. The workflow uses a different docker image (hosted on https://nexus.engageska-portugal.pt/) that has been created and maintained by the SKA System Team.

**NB: Docker image artefact will move to https://artefact.skatelescope.org/ in future!!!**

Some of the unit tests were reworked by mock patching the call to `helper_module.get_database` so that they wouldn't need a running instance of the tango-db server. With that in place, the Jenkinsfile was also update to use the `camtango_nodb_bionic` docker image.

Also added a workflow, `publish_to_pypi.yaml`, to automate the process of publishing the Python package to [PyPI](https://pypi.org/project/tango-simlib/) when a tag is created. For more information on the action used to upload to PyPI, see https://github.com/marketplace/actions/pypi-publish.

For more information on github actions, see [link](https://docs.github.com/en/actions/learn-github-actions).

*Screenshots or code snippets (if appropriate):*
  * N/A

*Definition of Done Checklist*

- [x] Code meets our [python style guidelines](https://docs.google.com/document/d/1aZoIyR9tz5rCWr2qJKuMTmKp2IzHlFjrCFrpDDHFypM/edit?usp=sharing)?
- [x] Unit tested (coded, passed, included)?
- [x] Requested at least 2 reviewers?
- [x] Commented code, particularly in hard-to-understand areas?
- [x] Made corresponding changes to the documentation (e.g. Python documentation, System Engineering Documentation, version description updates, README file, etc)?

JIRA: [N/A](https://skaafrica.atlassian.net/browse/JIRA_ISSUE_NUMBER)

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
   testing:
     runs-on: ubuntu-latest
     container:
-      image: nexus.engageska-portugal.pt/ska-docker/ska-python-buildenv:latest
+      image: nexus.engageska-portugal.pt/ska-docker/ska-python-buildenv:9.3.3.1
```
Comment (mmphego) [inline .github/workflows/tests.yaml:8]: 👍🏿 💯

**Comment by kmadisa** [comment]
- [ ] Add a static analysis step/job to ensure consistency with the Jenkins pipeline.

**Comment by kmadisa** [comment]
The https://nexus.engageska-portugal.pt/ service is currently down due to some upgrades (might take days).

- [ ] Re-run workflow.

**Comment by kmadisa** [comment]
See https://github.com/ska-sa/tango-simlib/runs/2771488752?check_suite_focus=true for linting failure test.

File: .github/workflows/unit_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Static Analysis & Unit Tests
+on: push
+
+jobs:
+  testing:
+    runs-on: ubuntu-latest
+    container:
+      image: nexus.engageska-portugal.pt/ska-docker/ska-python-buildenv:9.3.3.1
```
[the remark this answers is not in the document]
Comment (kmadisa) [reply .github/workflows/unit_tests.yaml:8]: Yes. It pulls it from the  https://nexus.engageska-portugal.pt/ artefact repository.

**Review by mmphego — APPROVED** [review]

**Pull request merged**